### DISCUSSION TITLE
Snark Accounts on Hypersdk

### DISCUSSION THREAD

**Discussion by manojkgorle** [ava-labs/hypersdk#941] [Show and Tell]
Hyper SDK supports adding any authentication type to a Hyper VM. The authentication type implements the `Auth` interface. 

Calling `Verify(ctx context.Context, msg []byte) error` returns nil for valid msg for tx data and an error in other cases.

Proving systems involving SNARKS will have a proving key and verifying key. Proving key is similar to private key, revealing it will help create fake proofs. Verifying key is similar to the public key, using this proofs are verified.

SNACS, aka Snark Accounts address, will be derived from the verifying key. Groth16 implementation from the Gnark team is used for snark accounts.

Read more: [https://x.com/ManojKGorle/status/1789745574754803930](https://x.com/ManojKGorle/status/1789745574754803930)
Github: [https://github.com/manojkgorle/hypersdk/tree/snac/examples/morpheusvm/README.md](https://github.com/manojkgorle/hypersdk/tree/snac/examples/morpheusvm/README.md)

**Comment by patrick-ogrady** [comment]
SUPER cool ❤️

**Comment by patrick-ogrady** [comment]
How long does it take to generate a simple proof for a transfer-only circuit?

**Reply by kpachhai** [reply]
Interested in this as well. @manojkgorle thoughts?
